packages/core/src/Stores/client-store.js

```diff
@@ -1819,7 +1819,9 @@ export default class ClientStore extends BaseStore {
 
                 this.root_store.ui.showAccountTypesModalForEuropean();
 
-                this.root_store.ui.toggleWelcomeModal({ is_visible: true, should_persist: true });
+                if (!this.is_uk) {
+                    this.root_store.ui.toggleWelcomeModal({ is_visible: true, should_persist: true });
+                }
```

Comment:
LGTM, but a minor suggestion
Is not being a UK user the only condition for NOT showing the welcome modal?
If this is the only condition we can keep it like this. 
but if there are other conditions in other places, too we can combine them in a new method like `shouldShowWelcomeModal` for better organization.

Reply:
Yes, this is the only condition for not showing and it's just for uk